### ISSUE TITLE
Helps about how to boostrap build of justbuild

### ISSUE THREAD

**Issue by topazus** [just-buildsystem/justbuild#44]
I used the following command to try to build it.
```bash
env PACKAGE=YES LOCALBASE=/usr \
    NON_LOCAL_DEPS='["google_apis","bazel_remote_apis"]' \
    python3 ./bin/bootstrap.py . build
```

see the error in build log: https://github.com/topazus/justbuild/actions/runs/7335489509/job/19973471274

**Comment by aehlig** [comment]
The `justbuild` bootstrap and package building process is designed for read-only source directories. Based on this assumption, the initial copying of the source tree to the build directory simply uses a recursive [tree copy](https://github.com/just-buildsystem/justbuild/blob/8d3d3a3dd8bda9f132dc3f15106c2f11bf22438b/bin/bootstrap.py#L383); this, however, fails if the destination, i.e., the work directory (or "build directory" in other terminology) is in the source tree. Using an out-of-source build directory (e.g., `../build` instead of `build` in the quoted command line) should fix it.

Do you have a use case where building in source is a strict requirement, or can we close this issue?

**Comment by aehlig** [comment]
BTW, an example of a working github workflow building just was written by @oreiche in his [fork](https://github.com/oreiche/justbuild/tree/github-ci/.github/workflows).

**Comment by aehlig** [comment]
Closing as resolved by 264ba60e221231109201da1c6d336c642959a0c2
